railgun/internal/proxy/clientgo_cached_proxy_resolver.go

```diff
@@ -203,12 +205,17 @@ func (p *clientgoCachedProxyResolver) startCacheServer() {
 				p.logger.Error(err, "object could not be deleted from the cache and will be discarded")
 				break
 			}
-		case <-syncTicker.C:
-			if err := p.updateKongAdmin(); err != nil {
+		case <-p.syncTicker.C:
+			updateConfigSHA, err := p.kongUpdater(p.ctx, p.lastConfigSHA, p.cache, p.ingressClassName, p.deprecatedLogger, p.kongConfig, p.enableReverseSync)
+			if err != nil {
 				p.logger.Error(err, "could not update kong admin")
 			}
+			p.lastConfigSHA = updateConfigSHA
 		case <-p.ctx.Done():
 			p.logger.Info("the proxy cache server's context is done, shutting down")
+			if err := p.ctx.Err(); err != nil {
```

Comment:
one more bit I've spotted. ctx.Done should stop the ticker, no?

Reply:
It might as well, sure. the server returns here, so in theory the ticket is going to get GC'd I think, but we can just stop it sooner rather than later as a micro-optimization. :+1:

Resolved by 3103189065d32e0e3cc1730f54c6fc0fd4692f0e